snmp/tests/test_e2e_core_profiles/test_profile_checkpoint_watchguard.py

```diff
@@ -0,0 +1,56 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import pytest
+
+from datadog_checks.dev.utils import get_metadata_metrics
+
+from .. import common
+from ..test_e2e_core_metadata import assert_device_metadata
+from .utils import (
+    assert_common_metrics,
+    assert_extend_generic_if,
+    create_e2e_core_test_config,
+    get_device_ip_from_config,
+)
+
+pytestmark = [pytest.mark.e2e, common.py3_plus_only, common.snmp_integration_only]
+
+
+def test_e2e_profile_checkpoint_watchguard(dd_agent_check):
+    config = create_e2e_core_test_config('checkpoint-watchguard')
+    aggregator = common.dd_agent_check_wrapper(dd_agent_check, config, rate=True)
+
+    ip_address = get_device_ip_from_config(config)
+    common_tags = [
+        'snmp_profile:checkpoint-watchguard',
+        'snmp_host:checkpoint-watchguard.device.name',
+        'device_namespace:default',
+        'snmp_device:' + ip_address,
+    ] + []
+
+    # --- TEST EXTENDED METRICS ---
+    assert_extend_generic_if(aggregator, common_tags)
+
```

Comment:
we're not checking for the `_generic-host-resources-base` extension here (the snmprec is missing those metrics which is why the `assert_all_metrics_covered` passes